storage/innobase/include/arch0arch.h

```diff
@@ -0,0 +1,1984 @@
+  Caller would use it to open and copy data from archived files.
+  @param[in]    idx             file index in the group
+  @param[out]   name_buf        file name and path. Caller must
+                                  allocate the buffer.
+  @param[in]    buf_len         allocated buffer length */
+  void get_file_name(uint idx, char *name_buf, uint buf_len)
+  {
+    ut_ad(name_buf != nullptr);
+
+    /* Build name from the file context. */
+    m_file_ctx.build_name(idx, m_begin_lsn, name_buf, buf_len);
+  }
+
+  /** Get the current file size for this group.
+  Fixed size files are used for archiving data in a group.
+  @return file size in bytes */
+  uint64_t get_file_size() const { return m_file_size; }
+
+  /** Get aligned start LSN for this group
+  @return aligned start LSN */
+  lsn_t get_first_lsn() const { return (m_first_lsn); }
+
+  /** Get start LSN for this group
+  @return start LSN */
+  lsn_t get_begin_lsn() const { return (m_begin_lsn); }
+
+  /** @return stop LSN for this group */
+  lsn_t get_end_lsn() const { return (m_end_lsn); }
+
+  /** @return stop block position of the group. */
+  Arch_Page_Pos get_stop_pos() const { return (m_stop_pos); }
+
+  /** @return uuid for the arch group */
+  Arch_group_uuid get_uuid() const { return m_uuid; }
+
+  /** Fetch the status of the page tracking system.
+  @param[out]   status  vector of a pair of (ID, bool) where ID is the
+  start/stop point and bool is true if the ID is a start point else false */
+  void get_status(std::vector<std::pair<lsn_t, bool>> &status)
+  {
+    m_file_ctx.get_status(status);
+
+    if (!is_active())
+      status.push_back(std::make_pair(m_end_lsn, false));
+  }
+
+  /** Open the file which was open at the time of a crash, during crash
+  recovery, and set the file offset to the last written offset.
+  @param[in]  write_pos   latest write position at the time of crash/shutdown
+  @param[in]  create_new  create new file if file not present
+  @return error code. */
+  dberr_t open_file(Arch_Page_Pos write_pos, bool create_new);
+
+  /** Align down LSN value to multiples of OS_FILE_LOG_BLOCK_SIZE with respect
+  to a reference LSN value.
+  @param[in] lsn LSN value to align
+  @param[in] ref_lsn reference LSN
+  @return aligned LSN value. */
+  static lsn_t align_lsn(lsn_t lsn, lsn_t ref_lsn)
+  {
+    lsn_t lsn_diff = (ref_lsn <= lsn) ? (lsn - ref_lsn) :
+        (ref_lsn - lsn + OS_FILE_LOG_BLOCK_SIZE);
+    lsn_diff = ut_uint64_align_down(lsn_diff, OS_FILE_LOG_BLOCK_SIZE);
+    lsn_t r = (ref_lsn <= lsn) ? (ref_lsn + lsn_diff) : (ref_lsn - lsn_diff);
+    ut_ad(r <= lsn);
+    return r;
+  }
+
+  /** Align down LSN value to multiples of OS_FILE_LOG_BLOCK_SIZE with respect
+  to groups first LSN value.
+  @param[in] lsn LSN value to align
+  @return aligned LSN value. */
+  lsn_t align_lsn(lsn_t lsn) const
+  {
+    ut_ad(m_first_lsn <= lsn);
+    return Arch_Group::align_lsn(lsn, m_first_lsn);
+  }
```

Comment:
No, we do not want to reintroduce the 512-byte log blocks after 685d958e38b825ad9829be311f26729cccf37c46 got rid of them.